osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/ImportHighScoresCommand.cs

```diff
@@ -575,14 +576,21 @@ private async Task<ScoreInfo> createReferenceScore(Ruleset ruleset, HighScore hi
                     scoreInfo.MaximumStatistics[HitResult.LegacyComboIncrease] = (int)maxComboAttribute.value - maxComboFromStatistics;
 #pragma warning restore CS0618
 
+                Beatmap beatmap = await connection.QuerySingleAsync<Beatmap>("SELECT * FROM osu_beatmaps WHERE `beatmap_id` = @BeatmapId", new
+                {
+                    BeatmapId = highScore.beatmap_id
+                }, transaction);
+
+                LegacyBeatmapConversionDifficultyInfo difficulty = LegacyBeatmapConversionDifficultyInfo.FromAPIBeatmap(beatmap.ToAPIBeatmap());
```

Comment:
Ascertaining that this construction works completely correctly (especially with respect to ruleset) is a really roundabout trip, but it does appear that this should do what is expected...